adapter/src/dbt/adapters/fal_experimental/utils/environments.py

```diff
@@ -111,14 +149,48 @@ def create_environment(name: str, kind: str, config: Dict[str, Any]):
             + ", ".join(REGISTERED_ENVIRONMENTS.keys())
         )
 
-    parsed_config = { key: val for key, val in config.items() if key not in CONFIG_KEYS_TO_IGNORE}
-
     if kind == "remote":
         parsed_config = _parse_remote_config(config, parsed_config)
 
     return env_type.from_config(parsed_config)
 
 
+def _build_hosted_env(
+        config: Dict[str, Any],
+        parsed_config: Dict[str, Any],
+        credentials: Any,
+        machine_type: str
+) -> BaseEnvironment:
+    from isolate_cloud.api import FalHostedServer
+    from isolate_cloud.api import CloudKeyCredentials
+    if not config.get("type"):
+        kind = "virtualenv"
+    else:
+        if config["type"] not in REMOTE_TYPES_DICT.keys():
+            raise RuntimeError(
+                f"Environment type {config['type']} is not supported. Supported types: {REMOTE_TYPES_DICT.keys()}")
+        kind = REMOTE_TYPES_DICT.get(config["type"])
+
+    if 'requirements' not in parsed_config.keys():
+        parsed_config['requirements'] = []
+
+    parsed_config['requirements'].append(f"dill=={importlib_metadata.version('dill')}")
```

Comment:
ATL: why do we have to add this one manually? 

1. I see we call to `_build_hosted_env` for local fal cloud with no `requirements`. So it would send a `dill==xxx` only requirements. Do we send the found packages in another env and put this one on top? I am a bit confused.

Reply:
Yes, locally the user can choose between pickle and dill but for isolate fal they have to use dill. I added it as an extra dependency now.

Also yes, we do layering in `run_in_environment_with_adapter` in adapter.py.